insights.go

```diff
@@ -21,29 +18,22 @@ func unregisterInsights() error {
 }
 
 func insightsIsRegistered() (bool, error) {
-	var errBuffer bytes.Buffer
-	cmd := exec.Command("/usr/bin/insights-client", "--status")
-	cmd.Stderr = &errBuffer
-
-	err := cmd.Run()
-
+	// While `insights-client --status` properly checks for registration status by
+	// asking Inventory, its two modes (legacy v. non-legacy API) behave
+	// differently (they return different texts with different exit codes) and
+	// we can't rely on the output or exit codes.
+	// The `.registered` file is always present on a registered system.
+	err := exec.Command("/usr/bin/insights-client", "--status").Run()
 	if err != nil {
-		// When the error is ExitError, then we know that insights-client only returned
-		// some error code not equal to zero. We do not care about error number.
-		var exitError *exec.ExitError
-		if errors.As(err, &exitError) {
-			// When stderr is not empty, then we should return this as error
-			// to be able to print this error in rhc output
-			stdErr := errBuffer.String()
-			if len(stdErr) == 0 {
```

Comment:
I would not delete old functionality altogether, because it will change behavior as you can see on this example:

```
root@localhost:/path/to/rhc# cat /usr/bin/insights-client 
#!/bin/bash

exit 1
root@localhost:/path/to/rhc# ./rhc status
Connection status for thinkpad-p1:

● Connected to Red Hat Subscription Management
● Cannot detect Red Hat Insights status: exit status 1
● The yggdrasil service is active

Manage your connected systems: https://red.ht/connector
```

If insights-client returns non-zero value, can we `return false, nil` to be consistent with old behavior?

Reply:
I wouldn't say that would be correct, because you get non-zero code even on network errors. It might have been the previous behavior, but it could be misleading. I would rather report the cached value risking it's out of date then report it's not connected to insights because the network glitched up.